mocha test/storage/aws-s3.js  testing actually creates a bucket in s3

<!--
1. If you have a question and not a bug/feature request please ask direct it to mailing list.
2. Please check if an issue already exists so there are no duplicates.
3. Check out and follow our Guidelines: https://github.com/cloudlibz/nodecloud/blob/master/CONTRIBUTING.md
4. Fill out the whole template so we have a good overview on the issue.
5. Do not remove any section of the template. If something is not applicable leave it empty but leave it in the Issue.
6. Please follow the template, otherwise we'll have to ask you to update it.
-->

# This is a Bug Report

## Description

* Unit testing storage with provider as aws, actually creates a bucket in the aws
* In the Unit test intercepting the http request to create a bucket and mock the http response

## Additional Data

* ***Operating System***: Linux mint 18.3 Cinnamon 64-bit
* ***Stack Trace***:
AWS S3
    1) should create S3 bucket
(node:10378) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): BucketAlreadyOwnedByYou: Your previous request to create the named bucket succeeded and you already own it.
(node:10378) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
    ✓ should not delete S3 bucket (1296ms)
    ✓ should create multipart upload (1055ms)
    ✓ should list all buckets (1337ms)
(node:10378) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): AssertionError: expected 'https://ncbucketcr.s3.us-west-2.amazonaws.com/key' to equal 'https://ncbucketcr.s3-us-west-2.amazonaws.com/key'
    2) should upload an arbitary sized buffer, blob, or stream


  3 passing (8s)
  2 failing

  1) AWS S3
       should create S3 bucket:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.


  2) AWS S3
       should upload an arbitary sized buffer, blob, or stream:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

--- a/test/storage/aws-s3.js
+++ b/test/storage/aws-s3.js
@@ -14,7 +14,7 @@ describe('AWS S3', () => {
   before(() => {});
 
   it('should create S3 bucket', (done) => {
-    nock('https://ncbucketcr.s3-us-west-2.amazonaws.com:443', {
+    nock('https://ncbucketcr.s3.us-west-2.amazonaws.com:443', {
       encodedQueryParams: true,
     })
       .put(
@@ -50,7 +50,7 @@ describe('AWS S3', () => {
   });
 
   it('should not delete S3 bucket', (done) => {
-    nock('https://ncbucketcr.s3-us-west-2.amazonaws.com:443', {
+    nock('https://ncbucketcr.s3.us-west-2.amazonaws.com:443', {
       encodedQueryParams: true,
     })
       .delete('/')
@@ -90,7 +90,7 @@ describe('AWS S3', () => {
   });
 
   it('should create multipart upload', (done) => {
-    nock('https://ncbucketcr.s3-us-west-2.amazonaws.com:443', {
+    nock('https://ncbucketcr.s3.us-west-2.amazonaws.com:443', {
       encodedQueryParams: true,
     })
       .post('/largeobject')
@@ -126,7 +126,7 @@ describe('AWS S3', () => {
   });
 
   it('should list all buckets', (done) => {
-    nock('https://s3-us-west-2.amazonaws.com:443', { encodedQueryParams: true })
+    nock('https://s3.us-west-2.amazonaws.com:443', { encodedQueryParams: true })
       .get('/')
       .reply(
         200,
@@ -146,7 +146,7 @@ describe('AWS S3', () => {
           'AmazonS3'
         ]
       );
-
+      
     s3.list({}).then((res) => {
       assert.typeOf(res.Buckets, 'array');
       done();
@@ -154,7 +154,7 @@ describe('AWS S3', () => {
   });
 
   it('should upload an arbitary sized buffer, blob, or stream', (done) => {
-    nock('https://ncbucketcr.s3-us-west-2.amazonaws.com:443', {
+    nock('https://ncbucketcr.s3.us-west-2.amazonaws.com:443', {
       encodedQueryParams: true,
     })
       .put('/key', 'ncunittest')
@@ -178,7 +178,7 @@ describe('AWS S3', () => {
     s3.upload(params).then((res) => {
       assert.equal(
         res.Location,
-        `https://${params.Bucket}.s3-us-west-2.amazonaws.com/key`
+        `https://${params.Bucket}.s3.us-west-2.amazonaws.com/key`
       );
       assert.equal(res.Bucket, params.Bucket);
       done();

